File: test/commands/hierarchy.test.js
import {
  runClangdQuery,
  assert,
  assertContains,
  assertMatches,
  countOccurrences,
  waitForDaemonReady,
} from '../test-helpers.js';

console.log('Testing hierarchy command...\n');

// Ensure daemon is ready before running tests
console.log('Waiting for daemon to be ready...');
await waitForDaemonReady();

// Test 1: View hierarchy of a derived class
console.log('Test 1: View hierarchy of Character class');
const result1 = await runClangdQuery(['hierarchy', 'Character']);
assert(result1.exitCode === 0, 'Command should succeed');
// Should show inheritance hierarchy
assertContains(result1.stdout, 'Inherits from:');
assertContains(result1.stdout, '└── GameObject');
assertContains(result1.stdout, 'Character - include/game/character.h');
assertContains(result1.stdout, '├── Enemy');
assertContains(result1.stdout, '└── Player');
console.log('✓ Test 1 passed\n');

console.log('All hierarchy tests passed! ✓');

File: go/test/hierarchy_test.go
package test

import (
	"testing"
)

func TestHierarchyCommand(t *testing.T) {
	tc := GetTestContext(t)
	
	t.Run("View hierarchy of Character derived class", func(t *testing.T) {
		result := tc.RunCommand("hierarchy", "Character")
		tc.AssertExitCode(result, 0)
		// Should show inheritance hierarchy
		tc.AssertContains(result.Stdout, "Inherits from:")
		tc.AssertContains(result.Stdout, "└── GameObject")
		tc.AssertContains(result.Stdout, "Character - include/game/character.h")
		tc.AssertContains(result.Stdout, "├── Enemy")
		tc.AssertContains(result.Stdout, "└── Player")
	})
	
	// Additional test: hierarchy of base class
	t.Run("View hierarchy of GameObject base class", func(t *testing.T) {
		result := tc.RunCommand("hierarchy", "GameObject")
		tc.AssertExitCode(result, 0)
		// Should show what GameObject inherits from
		tc.AssertContains(result.Stdout, "Inherits from:")
		tc.AssertContains(result.Stdout, "└── Updatable")
		// Should show derived classes
		tc.AssertContains(result.Stdout, "GameObject - include/core/game_object.h")
		tc.AssertContains(result.Stdout, "└── Character")
	})
	
	// Additional test: interface hierarchy
	t.Run("View hierarchy of Updatable interface", func(t *testing.T) {
		result := tc.RunCommand("hierarchy", "Updatable")
		tc.AssertExitCode(result, 0)
		// Should show classes that implement this interface
		tc.AssertContains(result.Stdout, "Updatable - include/core/interfaces.h")
		tc.AssertContains(result.Stdout, "GameObject")
	})
}